### DISCUSSION TITLE
TypeError: users.accountType is not a function [v4]

### DISCUSSION THREAD

**Discussion by punity-team** [adonisjs/core#1658] [Help]
I try to create a relationship between two tables.

One is users. They have a account type id for example 1 (free) as foreign key

One is account_types. They contain numbers like 1, 2, 3 like the primary key and then the role like free, pro etc.

Now I want to show the name of the role of it instead of the account_type_id

My Code:

**In the Model:**

`accountType () {
    return this.hasOne('App/Models/AccountType')
  }`

**In the Controller:**

`const users = User.query().limit(15).fetch() 
        return users.accountType()`

**When I execute:**

`TypeError: users.accountType is not a function`

**Comment by RomainLanz** [comment]
Hey @punity-team! 👋 

This is not how you load relationships for multiple models at the same time.
Please, read the documentation on how to eagerload or lazyload a relationship.

▶ https://adonisjs.com/docs/4.1/relationships#_eager_loading
▶ https://adonisjs.com/docs/4.1/relationships#_lazy_eager_loading

**Reply by punity-team** [reply]
Hey @RomainLanz 

I think this was the solution! 

Another question: How can I add a table prefix ? is this here working ?

https://github.com/adonisjs/lucid/issues/58

**Reply by punity-team** [reply]
@RomainLanz 

No, I get now this output:

`
{
        "id": "d704c04796fb44abac4ff6da2a0c8e",
        "username": "Jochen3",
        "accountType": {
            "id": 1,
            "name": "free"
        }
    }
`

I only want to have just the name of it like this:

`
{
        "id": "d704c04796fb44abac4ff6da2a0c8e",
        "username": "Jochen3",
        "accountType": "free"
        }
    }
`

or would you suggest to leave that so like later maybe when the role has special conditions or similar ?
